### PR TITLE
Update eth firehose filters

### PR DESCRIPTION
**Pull request by mangas**
Fixes https://github.com/graphprotocol/graph-node/issues/3606

### CODE REVIEW

**Comment by sduchesneau** [comment]
It works ! I'm seeing combined filters being used properly, I could trigger handlers from filters that would otherwise exclude each other, while processing ETH blocks with protobuf version 2

LGTM (I did not test POI regarding fixes in protobuf version 2, but this is not what is being addressed here)

File: chain/ethereum/proto/codec.proto [hunk 1]
```diff
@@ -74,105 +57,133 @@ message BlockRef {
...
 message BlockHeader {
-  bytes parent_hash = 1; /* geth: ParentHash + parentHash, parity: parentHash */
-  bytes uncle_hash = 2; /* geth: sha3Uncles, but sha3 is badly worded, so we prefer `uncle_hash`, parity: uncleHash */
-  bytes coinbase = 3; /* geth: Coinbase + miner, parity: coinbase */
-  bytes state_root = 4; /* geth: Root + json=stateRoot, parity: stateRoot */
-  bytes transactions_root = 5; /* geth: TxHash + transactionsRoot, parity: transactionsTrie */
-  bytes receipt_root = 6; /* geth: ReceiptHash + receiptRoot, parity: receiptTrie */
-  bytes logs_bloom = 7; /* internally called `Bloom`, parity uses `bloom`, geth's json uses `logsBloom` */
+  bytes parent_hash = 1;
+  // Uncle hash of the block, some refenrece it as `sha3Uncles`, but `sha3`` is badly worded, so we prefer `uncle_hash`
```
Comment (evaporei) [inline chain/ethereum/proto/codec.proto:61]: Typo: `refenrece` -> `reference`

File: chain/ethereum/proto/codec.proto [hunk 2]
```diff
@@ -74,105 +57,133 @@ message BlockRef {
...
 
+  // Type represents the Ethereum transaction type, available only since EIP-2718 & EIP-2930 activation which happened on Berlin fork.
+  // The value is always set even for transaction before Berlin fork because those before the fork are still legacy transactions.
+  Type type = 12;
+
+  enum Type {
+    // All transactions that ever existed prior Berlin fork before EIP-2718 was implemented.
+    TRX_TYPE_LEGACY = 0;
+
+    // Transaction that specicy an access list of contract/storage_keys that is going to be used
```
Comment (evaporei) [inline chain/ethereum/proto/codec.proto:133]: Typo: `specicy` -> `specify`

File: chain/ethereum/proto/codec.proto [hunk 3]
```diff
@@ -74,105 +57,133 @@ message BlockRef {
...
+    // Added in Berlin fork (EIP-2930).
+    TRX_TYPE_ACCESS_LIST = 1;
+
+    // Transaction that specifis an access list just like TRX_TYPE_ACCESS_LIST but in addition defines the
+    // max base gas gee and max priority gas fee to pay for this transaction. Transaction's of those type are
+    // executed against EIP-1559 rules which dictates a dynamic gas cost based on the congestion of the network.
+    TRX_TYPE_DYNAMIC_FEE = 2;
+  }
+
+  // AcccessList represents the storage access this transaction has agreed to do in which case those storage
```
Comment (evaporei) [inline chain/ethereum/proto/codec.proto:145]: Typo: `AcccessList` -> `AccessList`

**Review by evaporei — APPROVED** [review]
LGTM besides the typo comments and one question I had 🙂

File: chain/ethereum/src/codec.rs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #[rustfmt::skip]
-#[path = "protobuf/sf.ethereum.codec.v1.rs"]
+#[path = "protobuf/sf.ethereum.r#type.v1.rs"]
```
[the remark this answers is not in the document]
Comment (mangas) [reply chain/ethereum/src/codec.rs:2]: IIRC this is because the package name changed

**Pull request merged**